rust/settings/settings.bzl

```diff
@@ -247,6 +247,20 @@ def experimental_use_sh_toolchain_for_bootstrap_process_wrapper():
         build_setting_default = False,
     )
 
+def toolchain_linker_preference():
+    """A flag to control which linker is preferred for linking Rust binaries.
```

Comment:
can you expand this a bit to explain the difference between this setting and the cc_common.link one? i.e. what's the difference between enabling that and setting this one to `cc`? I'm worried that we're getting so many different linker options